aws/scenarios/vm/ec2instance/params.go

```diff
@@ -0,0 +1,83 @@
+package ec2instance
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/aws"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/agentinstall"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/common"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/os"
+)
+
+type Params struct {
+	ami                        string
+	instanceType               string
+	keyPair                    string
+	userData                   string
+	os                         os.OS
+	env                        aws.Environment
+	arch                       os.Architecture
+	optionalAgentInstallParams *agentinstall.Params
+}
+
+func newParams(env aws.Environment, options ...func(*Params) error) (*Params, error) {
+	params := &Params{
+		keyPair: "agent-ci-sandbox",
+		env:     env,
+	}
+
+	// By default use Ubuntu
+	return common.ApplyOption(params, WithOS(os.UbuntuOS, os.AMD64Arch), options)
```

Comment:
I don't think defaulting is required at this level. Using this means that scenario has been created entirely through code, not from a CLI-reachable scenario.
If some defaulting is required, it should be at a higher level.

Reply:
I think it is useful to be able to create a VM with simply `ec2instance.NewEc2Instance(ctx)` from the `datadog-agent` repository. What do you have in mind with higher level as this API is already high level?

Reply:
What I mean is something like:
```
// Function with params in input
NewEC2Instance(ctx, ...options)

// Shortcuts
NewEC2Ubuntu(ctx)
NewEC2Windows(ctx)
```

Reply:
I am keeping it for now as it doesn't hurt.